node/modules/storageminer.go

```diff
@@ -505,11 +505,11 @@ func (p *f3Participator) awaitLeaseExpiry(ctx context.Context, lease api.F3Parti
 			}
 			log.Errorw("Failed to check F3 progress while awaiting lease expiry. Retrying after backoff.", "attempts", p.backoff.Attempt(), "backoff", p.backoff.Duration(), "err", err)
 			p.backOff(ctx)
-		case progress.ID+2 >= lease.FromInstance+lease.ValidityTerm:
-			log.Infof("F3 progressed (%d) to within two instances of lease expiry (%d+%d). Renewing participation.", progress.ID, lease.FromInstance, lease.ValidityTerm)
+		case progress.ID+2 >= lease.ToInstance():
+			log.Infof("F3 progressed (%d) to within two instances of lease expiry (%d). Renewing participation.", progress.ID, lease.ToInstance())
 			return nil
 		default:
-			remainingInstanceLease := lease.ValidityTerm - progress.ID
+			remainingInstanceLease := lease.ToInstance() - progress.ID
```

Comment:
This was the bug. It would eventually start logging ~infinity instances remaining once we passed the max validity term.